plugins/infinispan51/src/main/java/org/radargun/service/Infinispan51Operations.java

```diff
@@ -74,7 +74,7 @@ public boolean remove(K key) {
       @Override
       public V getAndRemove(K key) {
          boolean startedTx = startTxAndLock(key);
-         boolean retval = super.remove(key);
```

Comment:
Seems like you found a bug - you should not call both super.remove and super.getAndRemove. Could you fix that? Also, I think that stopTx should be called in finally block.